### PR TITLE
Recover deleted application added

### PR DESCRIPTION
**Pull request by uihalagedara**

### CODE REVIEW

File: src/api/controllers/Application.controller.ts [hunk 1]
```diff
@@ -258,3 +258,18 @@ export const getDeletedApplicationsForAdmin = async (request: Request, response:
 			next();
 		});
 };
+
+/**
+ * recoverRemovedApplication
+ */
+export const recoverRemovedApplication = async (request: Request, response: Response, next: NextFunction) => {
+	await ApplicationService.recoverDeletedApplication(request.body.applicationId)
```
Comment (senuravihanjayadeva) [inline src/api/controllers/Application.controller.ts:266]: request.body.applicationId should be checked null or not in the controller

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

**Review by miyurugunarathna — APPROVED** [review]
LGTM

File: src/api/services/Application.service.ts [hunk 2]
```diff
@@ -274,3 +274,28 @@ export const getDeletedApplicationsForAdmin = async () => {
...
+			.then(async (application) => {
+				if (application) {
+					if (application.deletedAt !== null) {
+						application.deletedAt = null;
+
+						return application.save();
+					} else {
+						return { message: "This application is not deleted!", dateTime: new Date() };
+					}
+				}
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:293]: If there is no application in the else part `throw new Error("Application is not found");`

File: src/api/services/Application.service.ts [hunk 3]
```diff
@@ -274,3 +274,28 @@ export const getDeletedApplicationsForAdmin = async () => {
...
+*/
+export const recoverDeletedApplication = async (applicationId: string) => {
+	if (applicationId) {
+		return await ApplicationModel.findById(applicationId)
+			.then(async (application) => {
+				if (application) {
+					if (application.deletedAt !== null) {
+						application.deletedAt = null;
+
+						return application.save();
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:289]: This should be `return await application.save()`

**Review by rusiruavb — CHANGES_REQUESTED** [review]
@uihalagedara I have request some changes from your code. Please take a look.

**Review by senuravihanjayadeva — APPROVED** [review]

**Review by senuravihanjayadeva — APPROVED** [review]

**Pull request merged**